external/src/test/java/com/redhat/cloud/policies/engine/process/ReceiverTest.java

```diff
@@ -207,7 +204,9 @@ void testMoreComplexInput() throws Exception {
         testSubscriber.assertValueCount(1);
 
         Action action = deserializeAction(testSubscriber.values().get(0));
-        List<Map<String, String>> tags = (List<Map<String, String>>) action.getPayload().get("tags");
+        assertEquals(1, action.getEvents().size());
+
+        List<Map<String, String>> tags = (List<Map<String, String>>) action.getContext().get("tags");
```

Comment:
I don't see the `multiple values per tags key` thing you introduced being tested here. Could you improve this test a bit to cover that case?

Reply:
It was already there. I think is tested somewhere else but if not will add a test for it.

Reply:
Is tested in `testMoreComplexInput`.

The content there has the `location=Charmey` and `location=Neuchatel`

Reply:
Oh sorry, I missed that.